cloudberry/zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLGeneratortest.scala

```diff
@@ -0,0 +1,786 @@
+          |group by `hour`
+          | """.stripMargin.trim)
+    }
+
+    //28
+    "translate a text contain filter and group by geocell 10th" in {
+      val filter = Seq(textFilter)
+      val group = GroupStatement(Seq(byGeocell10), Seq(aggrCount))
+      val query = new Query(TwitterDataSetForSQL, Seq.empty, Seq.empty, filter, Seq.empty, Some(group), None)
+      val result = parser.generate(query, Map(TwitterDataSetForSQL -> twitterSchemaForSQL))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |select st_astext(Point(truncate(st_x(t.`coordinate`),1),truncate(st_y(t.`coordinate`),1)))  as `cell`,count(*) as `count`
+          |from `twitter_ds_tweet` t
+          |where match(t.`text`) against ('+zika +virus' in boolean mode)
+          |group by `cell`
+        """.stripMargin.trim)
+    }
+
+    //29
+    "translate a text contain filter and group by geocell 100th" in {
+      val filter = Seq(textFilter)
+      val group = GroupStatement(Seq(byGeocell100), Seq(aggrCount))
+      val query = new Query(TwitterDataSetForSQL, Seq.empty, Seq.empty, filter, Seq.empty, Some(group), None)
+      val result = parser.generate(query, Map(TwitterDataSetForSQL -> twitterSchemaForSQL))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |select st_astext(Point(truncate(st_x(t.`coordinate`),2),truncate(st_y(t.`coordinate`),2)))  as `cell`,count(*) as `count`
+          |from `twitter_ds_tweet` t
+          |where match(t.`text`) against ('+zika +virus' in boolean mode)
+          |group by `cell`
+        """.stripMargin.trim)
+    }
+
+    //30
+    "translate a text contain filter and group by geocell 1000th" in {
+      val filter = Seq(textFilter)
+      val group = GroupStatement(Seq(byGeocell1000), Seq(aggrCount))
+      val query = new Query(TwitterDataSetForSQL, Seq.empty, Seq.empty, filter, Seq.empty, Some(group), None)
+      val result = parser.generate(query, Map(TwitterDataSetForSQL -> twitterSchemaForSQL))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |select st_astext(Point(truncate(st_x(t.`coordinate`),3),truncate(st_y(t.`coordinate`),3)))  as `cell`,count(*) as `count`
+          |from `twitter_ds_tweet` t
+          |where match(t.`text`) against ('+zika +virus' in boolean mode)
+          |group by `cell`
+        """.stripMargin.trim)
+    }
+
+    //31
+    "translate a text contain filter and group by bin" in {
+      val filter = Seq(textFilter)
+      val group = GroupStatement(Seq(byBin), Seq(aggrCount))
+      val query = new Query(TwitterDataSetForSQL, Seq.empty, Seq.empty, filter, Seq.empty, Some(group), None)
+      val result = parser.generate(query, Map(TwitterDataSetForSQL -> twitterSchemaForSQL))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |select round(t.`geo_tag.stateID`/10)*10 as `state`,count(*) as `count`
```

Comment:
what does this `round` function do? the `query` doesn't ask for it. 
I guess it's for processing the `int`? is there any other easy way to handle the int case?